src/Raven.Server/Documents/Patch/BlittableOjectInstanceOperationScope.cs

```diff
@@ -0,0 +1,201 @@
+﻿using Sparrow.Json.Parsing;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Raven.Client;
+using System.Runtime.CompilerServices;
+using Jint.Native.Object;
+using Jint.Native;
+using Sparrow.Json;
+using System.Linq;
+
+namespace Raven.Server.Documents.Patch
+{
+    public class BlittableOjectInstanceOperationScope
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool ShouldFilterProperty(string property)
+        {
+            return property == Constants.Documents.Indexing.Fields.ReduceKeyFieldName ||
+                   property == Constants.Documents.Indexing.Fields.DocumentIdFieldName ||
+                   property == Constants.Documents.Metadata.Id ||
+                   property == Constants.Documents.Metadata.Etag ||
+                   property == Constants.Documents.Metadata.LastModified ||
+                   property == Constants.Documents.Metadata.IndexScore ||
+                   property == Constants.Documents.Metadata.ChangeVector ||
+                   property == Constants.Documents.Metadata.Flags;
+        }
+
+        private static readonly List<string> InheritedProperties = new List<string>
+        {
+            "length",
+            "Map",
```

Comment:
Do we need to properties? They are from the `RavenDB.js` file, but I think that JS already have replacements for all of those are they are all very C# centric.